### PR TITLE
improvement filter_select

### PR DESCRIPTION
**Pull request by kokspflanze**
- in `ZfcDatagrid\Filter` line 97 we use a string casted value, so i think we have to cast the value for the selected check too
- some codestyles changes
- some class fixxes in layout.phtml=)

Thanks =)

### CODE REVIEW

File: view/zfc-datagrid/renderer/bootstrapTable/layout.phtml [hunk 1]
```diff
@@ -144,26 +144,32 @@ if($this->rowClickAction !== null){
                         if ($col->isUserFilterEnabled() === true) {
 
                             if ($col->hasFilterSelectOptions() === true) {
-                                $htmlFilter = '<select name="toolbarFilters[' . $col->getUniqueId() . ']" style="width: 80%" onchange="this.form.submit()" class="form-control" >';
+                                $htmlFilter = sprintf(
+                                    '<select name="toolbarFilters[%s]" style="width: 80%%" onchange="this.form.submit()" class="form-control" >',
```
Comment (ThaDafinser) [inline view/zfc-datagrid/renderer/bootstrapTable/layout.phtml]: `width: 80%%` typo

File: view/zfc-datagrid/renderer/bootstrapTable/layout.phtml [hunk 2]
```diff
@@ -144,26 +144,32 @@ if($this->rowClickAction !== null){
...
                                 }
-                                $htmlFilter .= '</select>';
 
-                            } else if ($col->getType() instanceof \ZfcDatagrid\Column\Type\DateTime){
-                                $htmlFilter = '<input type="text" name="toolbarFilters[' . $col->getUniqueId() . ']" style="width: 80%" value="' . $col->getFilterActiveValue() . '" class="form-control daterange" />';
+                                $htmlFilter .= '</select>';
                             } else {
-                                $htmlFilter = '<input type="text" name="toolbarFilters[' . $col->getUniqueId() . ']" style="width: 80%" value="' . $col->getFilterActiveValue() . '" class="form-control" />';
+                                $htmlFilter = sprintf(
+                                    '<input type="text" name="toolbarFilters[%s]" style="width: 80%%" value="%s" class="form-control %s" />',
```
Comment (ThaDafinser) [inline view/zfc-datagrid/renderer/bootstrapTable/layout.phtml]: `width: 80%%`typo

**Comment by ThaDafinser** [comment]
@kokspflanze looks fine, expect the 2 typos. Can you fix them? Then i'll merge it

**Comment by kokspflanze** [comment]
the %% is a workflow for thw sprintf the show a % =]

**Comment by ThaDafinser** [comment]
Didnt know that thank you! http://stackoverflow.com/questions/3666734/php-sprintf-escaping

**Pull request merged**